<p align="center"><img src='https://i.postimg.cc/fLjL5q7K/Schermata-2022-06-12-alle-21-52-50.png' height=180px/></p>

## VariantHunter: *a tool for fast detection of emerging SARS-CoV-2 variants*

### Introduction

While over 10 million genomic sequences of SARS-CoV-2 are currently available through dedicated repositories, the identification of novel viral variants is engaging a huge number of virologists worldwide. We designed VariantHunter, which analyzes the frequencies of amino acid mutations over 4-week spans in specific locations and lineages, in order to hunt novel emerging variants as early as possible. The two types of supported analysis (lineage independent and lineage specific) allow user- friendly monitoring of the viral evolution, empowering genomic surveillance.

---

### What is Variant Hunter

VariantHunter is a highly flexible and user-friendly tool for the systematic monitoring of the evolution of SARS-CoV-2 both at global and regional level. VariantHunter makes use of simple statistics and visual representations (plots and heatmaps) to track viral evolution by highlighting the increase or decrease of prevalence of a given amino acid changes or group of amino acid changes (for example those defining a new lineage); analyses are executed by selecting a 4-week time period and an arbitrary geographical area (at the levels of continent, country, and region). A "data explorer" utility can be used to explore the availability of data and refine the selection of time intervals. Additionally, amino acid changes characteristic of a lineage can be highlighted to facilitate the discovery of novel variants. The tool operates on aggregated data and access to viral genome sequences is not required


<b>
  
[> More information about the tool](http://gmql.eu/variant_hunter/about)

[> Go to the online version of the tool](http://gmql.eu/variant_hunter)

[> How to run it using Docker](http://gmql.eu/variant_hunter/about)
 
</b>

---
